decidim-core/app/assets/javascripts/decidim/vizzs/metrics.js.es6

```diff
@@ -1,26 +1,23 @@
-/* eslint-disable no-unused-vars */
-
-// Outside of the closure to make it public
-let DATACHARTS = null;
-
-const fetchDatacharts = () => {
-
-  const metrics = [{
-    key: "NAME_TO_BE_IN_THE_HTML-1",
-    query: "GRAPHQL_QUERY-1"
-  }, {
-    key: "NAME_TO_BE_IN_THE_HTML-2",
-    query: "GRAPHQL_QUERY-2"
-  }]
-
-  const fetch = (query) => $.post("<-- GRAPHQL_URL -->", query)
-
-  const promises = metrics.map((metric) => fetch(metric.query).then((response) => {
-    DATACHARTS[metric.key] = response
-
-    return DATACHARTS
-  }))
-
-  Promise.all(promises).then(() => DATACHARTS)
-
-}
+/* global areachart */
+
+$(() => {
+  const query = (metric) => {
+    return {query: `{ ${metric} { metric { key value } } }`};
+  }
+
+  const fetch = (metric) => $.post("api", query(metric))
+
+  $(".metric-chart:visible").each((index, container) => {
+    let metric = $(container).data("metric");
+    let info = $(container).data("info");
+    fetch(metric).then((response) => {
+      let data = response.data[metric].metric || {};
+      areachart({
```

Comment:
If you call straight to the areachart, we're creating a dependency between the charts and the metrics. Take a look at [this PR](https://github.com/decidim/decidim/blob/7a81c28a4d273736945eea52340ed503fe89a26c/decidim-core/app/assets/javascripts/decidim/vizzs/orgchart.js.es6) to see how is implemented. Also [this render function](https://github.com/decidim/decidim/blob/7a81c28a4d273736945eea52340ed503fe89a26c/decidim-core/app/assets/javascripts/decidim/vizzs/renders.js.es6).

I'd say it's OK for this PR, because it's so huge. But you know there's gonna be more graphics we must deal with. Please, create another issue to "unify" a way to render all datacharts.